src/components/Pagination/Pagination.jsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import './Pagination.scss';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+class Pagination extends React.Component {
+  state = {
+    maxCountPage: this.props.maxCount,
+  };
+
+  createShowPages = (pagesNumber) => {
+    const { selectedPage: selPage } = this.props;
+    const pagesForShow = [];
+
+    if (pagesNumber === 1) {
+      return [1];
+    }
+
+    pagesForShow.push(1);
+    for (let i = 2; i < pagesNumber; i += 1) {
+      if (i === selPage || i - 1 === selPage || i + 1 === selPage) {
+        pagesForShow.push(i);
+      } else if (pagesForShow[pagesForShow.length - 1] !== null) {
+        pagesForShow.push(null);
+      }
+    }
+
+    pagesForShow.push(pagesNumber);
+
+    return pagesForShow;
+  };
+
+  changePageHandler = (newSelectedPage) => {
+    const { onChangeSelected, selectedPage } = this.props;
+
+    if (selectedPage !== newSelectedPage) {
+      onChangeSelected(newSelectedPage);
+    }
+  };
+
+  maxCountHandler(event) {
+    const { value } = event.target;
+    const { maxCount, onChangeMaxCount } = this.props;
+
+    if (maxCount !== value) {
+      onChangeMaxCount(+value);
+      this.setState({ maxCountPage: value });
```

Comment:
`Pagination` component should be stateless.
This handler should change the state of the `App` that in return provides different prop to `Pagination`.